clients/drcachesim/common/trace_entry.h

```diff
@@ -434,6 +434,14 @@ typedef enum {
      */
     TRACE_MARKER_TYPE_RSEQ_ENTRY,
 
+    /**
+     * This marker is emitted prior to each system call invocation, after the
+     * instruction fetch entry for the system call gateway instruction from user mode. The
+     * marker value contains the system call number.  If these markers are present, the
+     * file type #OFFLINE_FILE_TYPE_SYSCALL_NUMBERS is set.
+     */
+    TRACE_MARKER_TYPE_SYSCALL,
```

Comment:
Not sure I know what you mean: a note about what that needs to be fixed?

Reply:
TRACE_MARKER_TYPE_SYSCALL_IDX  is written in event_post_syscall. The same two scenarios that we try to handle for TRACE_MARKER_TYPE_SYSCALL might apply there too?

Reply:
Side question: why in post instead of pre?  There is no post for some syscalls; but more importantly this may result in these _IDX markers not being near the syscall instructions or number markers since a blocking syscall may cause a thread switch in between the pre and post.  Wouldn't we want to insert the PT trace at pre-syscall?

For the same scenarios: yes, but the solutions for the number marker should solve the issue and we won't need any extra work for these _IDX markers I would think (except the post issue above; I thought these were in pre right after the number marker -- I guess that's the PT start call is all).

Reply:
Added a comment in event_post_syscall that A) we probably want to move it to pre and B) the signal issues hit by the number marker affect it too but our fixes should fix for both.